application/app/Http/Requests/UpdateInstitutionRequest.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace App\Http\Requests;
+
+use App\Rules\PhoneNumberRule;
+use Illuminate\Contracts\Validation\ValidationRule;
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Validation\Validator;
+
+class UpdateInstitutionRequest extends FormRequest
+{
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, ValidationRule|array|string>
+     */
+    public function rules(): array
+    {
+        return [
+            'name' => [
+                'string',
+                'filled',
+            ],
+            'phone' => [
+                'nullable',
+                'string',
+                new PhoneNumberRule,
+            ],
+            'email' => [
+                'nullable',
+                'string',
+                'email',
+            ],
+            'short_name' => [
+                'nullable',
+                'string',
+                'max:3',
+            ],
+        ];
+    }
+
+    public function after(): array
+    {
+        return [
+            function (Validator $validator): void {
+                if (! $this->hasAny(['name', 'phone', 'email', 'short_name'])) {
+                    $validator->errors()->add('.', 'At least one field to update is required.');
```

Comment:
I feel that this is a little overthinking/-complicating and is not really relevant, since it does not exactly check if the underlying data changed or not. Otherwise we should add this to every possible data "write" endpoint to verify it and be consistent in this approach. Better way would be to add this as a global functionality as a middleware, but there may be cases where some object needs to be created without input parameters.

I don't see a problem in this by letting it happen. If the reason behind this is by denying ddosing some endpoint, then rate limiting can be applied. Same case would happen when querying "read" endpoints

Reply:
> it does not exactly check if the underlying data changed or not

That’s a good point.

> Otherwise we should add this to every possible data "write" endpoint to verify it and be consistent in this approach.

I don’t think having the validation insome endponts necessarily means we have to put it everywhere. But I agree that having it in only a couple endpoints would be inconsistent.

I’ll remove this validation check and associated test cases.